### ISSUE TITLE
prefer-const should not work on array destructuring assignment

### ISSUE THREAD

**Issue by CapacitorSet** [eslint/eslint#7429]
**Tell us about your environment**

* **ESLint Version:** v2.13.1
* **Node Version:** v6.9.1
* **npm Version:** v3.10.8

**What parser (default, Babel-ESLint, etc.) are you using?** `babel-eslint`

**Please show your full configuration:**

```js
module.exports = {
    "parser": "babel-eslint",
    "extends": "google",
    "rules": {
        "camelcase": 1,
        "curly": 0,
        "eol-last": 0,
        "indent": ["error", 4],
        "max-len": 0,
        "new-cap": 1,
        "no-extend-native": 0,
        "no-loop-func": 0,
        "no-unused-vars": 1,
        "no-var": 1,
        "prefer-const": 1,
        "require-jsdoc": 0
    }
};
```

**What did you do? Please include the actual source code causing the issue.**

```js
        let type;
        let pluginName;
        let property;
        let jsonValue;
        [type, pluginName, property, ...jsonValue] = args;
```

**What did you expect to happen?** I expected ESLint not to display any warning.

**What actually happened? Please include the actual, raw output from ESLint.**

It reports the following warnings:

```
  20:10  warning  'type' is never reassigned, use 'const' instead        prefer-const
  20:16  warning  'pluginName' is never reassigned, use 'const' instead  prefer-const
  20:28  warning  'property' is never reassigned, use 'const' instead    prefer-const
```

However, if one changes the `let`s to `const`s, running ESLint again gives:

```
  16:19  error  Parsing error: Unexpected token
```

**Comment by not-an-aardvark** [comment]
I think instead of declaring the variables separately, you can just use this:

```js
const [type, pluginName, property, ...jsonValue] = args;
```

**Comment by kaicataldo** [comment]
Closing this issue as it looks like the question has been answered. Please feel free to visit us in the [ESLint Gitter](https://gitter.im/eslint/eslint) if you have any other issues!
